purchase_stock_price_variance/models/stock_picking.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2025 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+
+
+class Stockpick(models.Model):
+    _inherit = "stock.picking"
+
+    bypass_price_variance_check = fields.Boolean(
+        copy=False,
+        tracking=True,
+        groups="purchase_stock_price_variance.group_bypass_price_variance_check",
+        help="If enabled, no error is raised for price variance between "
+        "the product's standard price and purchase receipt unit price.",
+    )
+
+    def _action_done(self):
+        global_price_variance_threshold_percent = (
+            self.env.company.price_variance_threshold_percent
+        )
+        global_price_variance_threshold_amount = (
+            self.env.company.price_variance_threshold_amount
+        )
+        for pick in self:
+            if pick.picking_type_id.code != "incoming":
+                continue
+            if pick.sudo().bypass_price_variance_check:
+                continue
+            for move in pick.move_ids_without_package:
+                product = move.product_id
+                if product.bypass_price_variance_check:
+                    continue
+                threshold_percent = (
+                    product.price_variance_threshold_percent
+                    or global_price_variance_threshold_percent
+                )
+                threshold_amount = (
+                    product.price_variance_threshold_amount
+                    or global_price_variance_threshold_amount
+                )
+                received_price = move.price_unit
```

Comment:
I think it should be safer this way to update the cost based on the current exchange rate.
```suggestion
                received_price = move._get_price_unit()
```